clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/IndexWebServiceSpec.scala

```diff
@@ -0,0 +1,137 @@
+package org.broadinstitute.clio.server.webservice
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.{MalformedRequestContentRejection, Route}
+import io.circe.Json
+import org.broadinstitute.clio.server.dataaccess.MemorySearchDAO
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.{
+  ClioDocument,
+  ElasticsearchIndex
+}
+import org.broadinstitute.clio.server.service.MockIndexService
+import org.broadinstitute.clio.transfer.model.TransferIndex
+import org.broadinstitute.clio.util.model.UpsertId
+
+abstract class IndexWebServiceSpec[
+  TI <: TransferIndex,
+  D <: ClioDocument: ElasticsearchIndex
+] extends BaseWebserviceSpec {
+
+  val memorySearchDAO = new MemorySearchDAO()
+
+  def webServiceName: String
+  val mockService: MockIndexService[TI, D]
+  val webService: IndexWebService[TI, D]
+  val onPremKey: webService.indexService.transferIndex.KeyType
+  val cloudKey: webService.indexService.transferIndex.KeyType
+  val badMetadataMap: Map[String, String]
+  val badQueryInputMap: Map[String, String]
```

Comment:
Can you make all these `def`s?

Reply:
Looks like I can't access members of webService if I make it a def, so leaving that one